tests/scripts/all.sh

```diff
@@ -483,6 +483,15 @@ if_build_succeeded env OPENSSL_CMD="$OPENSSL_LEGACY" tests/compat.sh -m 'ssl3'
 msg "build: SSLv3 - ssl-opt.sh (ASan build)" # ~ 6 min
 if_build_succeeded tests/ssl-opt.sh
 
+msg "build: Default + SSLv3 + SSLv2 + MBEDTLS_DEPRECATED_REMOVED" # ~30s
+cleanup
+cp "$CONFIG_H" "$CONFIG_BAK"
+scripts/config.pl set MBEDTLS_SSL_PROTO_SSL3
+scripts/config.pl set MBEDTLS_SSL_SRV_SUPPORT_SSLV2_CLIENT_HELLO
+scripts/config.pl set MBEDTLS_DEPRECATED_REMOVED
+CC=gcc cmake .
+make
```

Comment:
Well this test is correct as the code stands, but I think we should actually fail to build for this particular config (and don't need a test for that: we don't usually check that everything that `check_config.h` forbids actually fails).